apps/website/src/components/navigation/floating-menu-mobile.tsx

```diff
@@ -0,0 +1,115 @@
+import { useEffect, useRef, useState } from 'react'
+
+import { Button, IconButton } from '@status-im/components'
+import { CloseIcon, DownloadIcon, MenuIcon } from '@status-im/icons'
+import { cx } from 'class-variance-authority'
+
+import { useLockScroll } from '@/hooks/use-lock-scroll'
+import { useOutsideClick } from '@/hooks/use-outside-click'
+
+import { Link } from '../link'
+import { Logo } from '../logo'
+import { AccordionMenu } from './accordion-menu'
+
+const FloatingMenuMobile = () => {
+  const [visible, setVisible] = useState(false)
+  const [open, setOpen] = useState(false)
+  const openRef = useRef(open)
+  openRef.current = open
+
+  // Using ref to prevent re-running of useEffect
+  const visibleRef = useRef(visible)
+  const scrollYRef = useRef(0)
+  visibleRef.current = visible
+
+  // Close menu on outside click
+  const ref = useOutsideClick(() => setOpen(false))
+  useLockScroll(open)
+
+  useEffect(() => {
+    const handleScroll = () => {
+      const isMenuOpen = openRef.current
+      const isScrollingUp = window.scrollY < scrollYRef.current
+      const detectionPoint = window.scrollY > 60
+
+      if (detectionPoint && isScrollingUp) {
+        if (!visibleRef.current) {
+          setVisible(true)
+        }
+      } else {
+        if (!isMenuOpen) {
+          setVisible(false)
+        }
+      }
+      scrollYRef.current = window.scrollY
+    }
+
+    const isDesktop = window.innerWidth > 768
+
+    if (isDesktop) return
+    handleScroll()
+
+    window.addEventListener('scroll', handleScroll, { passive: true })
+    return () => {
+      window.removeEventListener('scroll', handleScroll)
+    }
+  }, [])
```

Comment:
Have you considered using [useScroll](https://www.react-spring.dev/docs/utilities/use-scroll) hook from `@react-spring/web`? It is better optimized because it keeps it out of React state.